Palaso.MSBuildTasks/UnixName.cs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2014 SIL International
+// This software is licensed under the MIT License (http://opensource.org/licenses/MIT)
+// Parts based on code by MJ Hutchinson http://mjhutchinson.com/journal/2010/01/25/integrating_gtk_application_mac
+
+using System;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+
+namespace Palaso.BuildTasks
+{
+	/// <summary>
+	/// Determines the Unix Name of the operating system executing the build.
+	/// This is useful when determining Mac vs Linux during a build.
+	/// On Mac, the output Value will be "Darwin".
+	/// On Linux, the output Value will be "Linux".
+	/// 
+	/// This can be used to set DefineConstants during the PreBuild Target.
+	/// here is an example build/platform.targets file that can be included
+	/// in a CSPROJ file.  SYSTEM_OSX or SYSTEM_LINUX will be defined and 
+	/// can be used in the C# code for #if conditional compilation.
+	/// <?xml version="1.0" encoding="utf-8" ?>
+	/// <Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+	///   <UsingTask TaskName="UnixName" AssemblyFile="Palaso.BuildTasks.dll" />
+	///   <Target Name="BeforeBuild">
+	/// 	<UnixName>
+	///       <Output TaskParameter="Value" PropertyName="UNIX_NAME" />
+	///     </UnixName>
+	/// 	<PropertyGroup>
+	///       <DefineConstants Condition="'$(UNIX_NAME)' == 'Darwin'">$(DefineConstants);SYSTEM_OSX</DefineConstants>
+	///       <DefineConstants Condition="'$(UNIX_NAME)' == 'Linux'">$(DefineConstants);SYSTEM_LINUX</DefineConstants>
```

Comment:
Apple refers to its operating system as OSX and to the hardware as Mac.  It dropped MacOSX a while ago.  So for consistency, I could change Palaso.PlatformUtilities.Platform.IsMac to .IsOSX.  Or we could use DARWIN everywhere. :-)

FYI, SYSTEM_OSX is already being used in Palaso.Tests and I need to fix that place.  

http://www.apple.com/osx/